### ISSUE TITLE
ua disable fips doesn't work in ua client 27

### ISSUE THREAD

**Issue by davecore82** [canonical/ubuntu-advantage-client#1389]
I'm trying to disable FIPS from an Ubuntu Pro FIPS 18.04 image in AWS. I updated to the latest ua client in the daily PPA. I have a prompt to disable it but it fails:

```
ubuntu@ip-172-31-60-238:~$ sudo add-apt-repository ppa:canonical-server/ua-client-daily

ubuntu@ip-172-31-60-238:~$ sudo apt install ubuntu-advantage-pro ubuntu-advantage-tools

ubuntu@ip-172-31-60-238:~$ ua version
27.0-945~gedf4a7e~ubuntu18.04.1

ubuntu@ip-172-31-60-238:~$ ua status
SERVICE ENTITLED STATUS DESCRIPTION
cis-audit no — Center for Internet Security Audit Tools
esm-infra yes enabled UA Infra: Extended Security Maintenance
fips yes enabled NIST-certified FIPS modules
fips-updates no — Uncertified security updates to FIPS modules
livepatch yes n/a Canonical Livepatch service
[...]

ubuntu@ip-172-31-60-238:~$ sudo ua disable fips
This will disable access to certified FIPS packages.
Are you sure? (y/N) y
Could not enable FIPS.

ubuntu@ip-172-31-60-238:~$ ua status
SERVICE ENTITLED STATUS DESCRIPTION
cis-audit no — Center for Internet Security Audit Tools
esm-infra yes enabled UA Infra: Extended Security Maintenance
fips yes enabled NIST-certified FIPS modules
fips-updates no — Uncertified security updates to FIPS modules
livepatch yes n/a Canonical Livepatch service
[...]

```
I tried rebooting after but I'm still running the fips kernel and fips is enabled:

```
ubuntu@ip-172-31-60-238:~$ uname -a
Linux ip-172-31-60-238 4.15.0-2000-aws-fips #4-Ubuntu SMP Tue Jan 28 12:41:43 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

ubuntu@ip-172-31-60-238:~$ cat /proc/sys/crypto/fips_enabled
1
```

**Comment by mitechie** [comment]
Thanks, we've got some manual instructions to help but especially with a FIPS pro image that's built and shipped with the single FIPS kernel it's a risky change that we'd prefer the user/operator be involved with. 

https://discourse.ubuntu.com/t/ubuntu-advantage-disabling-fips-manually/20738/4

We'll work in the future to help smooth this out as we advance the client and so keep this bug to help us discuss the work required to make this smooth, automatic, and bombproof.

**Comment by blackboxsw** [comment]
Thanks again @davecore82.  As @mitechie  mentioned, disabling FIPS on a FIPS PRO image is a recipe for risk because the image was built with only one kernel.

TLDR: to attempt to get to a "PRO" vm with an upgraded ubuntu-advantage-tools (and not FIPS), probably just launch the https://ubuntu.com/aws/pro image and upgrade ua-tools and pro
```bash
sudo add-apt-repository ppa:canonical-server/ua-client-daily
sudo apt-get update
sudo apt-get install ubuntu-advantage-tools ubuntu-advantage-pro
```
OR more safely use `sudo add-apt-repository ppa:uaclient/stable` which is our released stable versions of ubuntu-advantage-tools/pro (version 26.1)


This issue represents two problems:
   1. The error message emitted is incorrect "Could not enable FIPS." should be Could not disable FIPS" (easy doc fix) which we can tie to this issue as it gets resolved
   2.  Failure in logs about inability `apt-get remove linux-aws-fips libssl etc` because the  current contract for AWS Ubuntu PRO FIPS images does not represent the `ubuntu-aws-fips` metapkg as the package to install/remove during `ua enable/disable` operations

Part 2 is known and will not work until we schedule the publishing of AWS Ubuntu PRO FIPS images with u-a-tools and u-a-pro 26.1 included. (expected 1/2 weeks) and a lock-step update of the AWS contract related to FIPS PRO.

Two things that need to happen to fix part 2:
 
1: Publish of AWS PRO FIPS images containing 26.1 or later
2: A lockstep contract change to represent a new ubuntu-aws-fips metapkg  as the deb package which gets installed/removed during `ua enable/disable` operations

**Comment by blackboxsw** [comment]
Part 2 has just been published/remedied as of 06/11/2021. All Ubuntu PRO and Ubuntu PRO FIPS images and contracts have been updated for AWS and Azure PRO image offerings. `ua disable fips` will now only remove ubuntu-fips, ubuntu-aws-fips or ubuntu-azure-fips metapackages which effectively just disables "FIPS-mode" instead of removing openssl/libssl etc making the system unusable.
